runtime/serde/serde-xml/common/src/aws/smithy/kotlin/runtime/serde/xml/XmlDeserializer.kt

```diff
@@ -213,65 +214,78 @@ internal class XmlListDeserializer(
  * @param parentToken initial token of associated structure
  * @param parsedFieldLocations list of [FieldLocation] representing values able to be loaded into deserialized instances
  */
-internal class XmlStructDeserializer(
+private class XmlStructDeserializer(
     private val objDescriptor: SdkObjectDescriptor,
-    private val reader: XmlStreamReader,
+    reader: XmlStreamReader,
     private val parentToken: XmlToken.BeginElement,
     private val parsedFieldLocations: MutableList<FieldLocation> = mutableListOf(),
+    private val unwrapped: Boolean,
 ) : Deserializer.FieldIterator {
     // Used to track direct deserialization or further nesting between calls to findNextFieldIndex() and deserialize<Type>()
     private var reentryFlag: Boolean = false
 
+    private val reader: XmlStreamReader = if (unwrapped) reader else reader.subTreeReader(XmlStreamReader.SubtreeStartDepth.CHILD)
+
     override fun findNextFieldIndex(): Int? {
-        if (inNestedMode()) {
-            // Returning from a nested struct call.  Nested deserializer consumed
-            // tokens so clear them here to avoid processing stale state
-            parsedFieldLocations.clear()
-        }
+        if (unwrapped) {
+            val FIELD_INDEX: Int = 0
```

Comment:
**style**: Move this to a constant in the file e.g. `private const val FIRST_FIELD_INDEX: Int = 0`